packages/plugin-ext-vscode/src/node/plugin-vscode-file-handler.ts

```diff
@@ -41,6 +42,13 @@ export class PluginVsCodeFileHandler implements PluginDeployerFileHandler {
 
         const unpackedPath = path.resolve(this.unpackedFolder, path.basename(context.pluginEntry().path()));
         await context.unzip(context.pluginEntry().path(), unpackedPath);
+        if (context.pluginEntry().path().endsWith('.tgz')) {
+            const extensionPath = path.join(unpackedPath, 'package');
+            const vscodeNodeModulesPath = path.join(extensionPath, 'vscode_node_modules.zip');
```

Comment:
built-in extensions are already republished with zip for node modules: https://github.com/theia-ide/vscode-builtin-extensions/commit/51e5e2797a110bdfe663a820fef1546a215d6642#diff-1b65ea7e7a9dd6aa50436ecdd461f0aaR23